src/native/corehost/browserhost/libBrowserHost.js

```diff
@@ -0,0 +1,536 @@
+//! Licensed to the .NET Foundation under one or more agreements.
+//! The .NET Foundation licenses this file to you under the MIT license.
+//! This is generated file, see src/native/libs/Browser/rollup.config.defines.js
+
+
+var libBrowserHost = (function (exports) {
+    'use strict';
+
+    // Licensed to the .NET Foundation under one or more agreements.
+    // The .NET Foundation licenses this file to you under the MIT license.
+    let Module;
+    let runtimeApi;
+    let Logger = {};
+    let Assert = {};
+    let JSEngine = {};
+    let loaderExports = {};
+    let runtimeExports = {};
+    let nativeExports = {};
+    let interopExports = {};
+    let dotnetInternals;
+    function getInternals() {
+        return dotnetInternals;
+    }
+    function setInternals(internal) {
+        dotnetInternals = internal;
+        runtimeApi = dotnetInternals.runtimeApi;
+        Module = dotnetInternals.runtimeApi.Module;
+    }
+    function updateInternals() {
+        if (dotnetInternals.updates === undefined) {
+            dotnetInternals.updates = [];
+        }
+        for (const updateImpl of dotnetInternals.updates) {
+            updateImpl();
+        }
+    }
+    function updateInternalsImpl() {
+        if (Object.keys(loaderExports).length === 0 && dotnetInternals.loaderExportsTable) {
+            loaderExports = {};
+            Logger = {};
+            Assert = {};
+            JSEngine = {};
+            loaderExportsFromTable(dotnetInternals.loaderExportsTable, Logger, Assert, JSEngine, loaderExports);
+        }
+        if (Object.keys(runtimeExports).length === 0 && dotnetInternals.runtimeExportsTable) {
+            runtimeExports = {};
+            runtimeExportsFromTable(dotnetInternals.runtimeExportsTable, runtimeExports);
+        }
+        if (Object.keys(nativeExports).length === 0 && dotnetInternals.hostNativeExportsTable) {
+            nativeExports = {};
+            hostNativeExportsFromTable(dotnetInternals.hostNativeExportsTable, nativeExports);
+        }
+        if (Object.keys(interopExports).length === 0 && dotnetInternals.interopJavaScriptNativeExportsTable) {
+            interopExports = {};
+            interopJavaScriptNativeExportsFromTable(dotnetInternals.interopJavaScriptNativeExportsTable, interopExports);
+        }
+    }
+    /**
+     * Functions below allow our JS modules to exchange internal interfaces by passing tables of functions in known order instead of using string symbols.
```

Comment:
Why? Feels fragile. Is it mandatory for it to work this way?

Reply:
I think it's less fragile than maintaining the explicit list of symbols.
The complexity here is around JS name mangling/minification in Release mode.

In any case, this is how emscripten linker works. They execute JS code of your library and then call `toString()` on functions that they want to link. It's nasty.

See [javascript-limits-in-library-files](https://emscripten.org/docs/porting/connecting_cpp_and_javascript/Interacting-with-code.html#javascript-limits-in-library-files)

I will add this link